backend/src/utils/getPort.ts

```diff
@@ -0,0 +1,54 @@
+import { Server } from 'http';
+
+/**
+ * Set the port for the server to listen on based on the environment.
+ * 
+ * @returns A promise that resolves to the port number.
+ */
+export default async function getPort(): Promise<number> {
```

Comment:
This is an interesting way of doing this. I think it can work, but I'm also wondering why you decided to go with this over letting express choose a random unused port by passing in 0 for the port number (https://expressjs.com/en/4x/api.html#app.listen)?

I asked an LLM how to get a port number that way and this is what it gave me:
```
const server = app.listen(() => {
  const address = server.address();
  if (address && typeof address !== 'string') {
    const port = address.port;
    console.log(`Server is running on port ${port}`);
  }
});
```

Reply:
I have no strong reason to do it this way, I think I just got carried away. 😁 Thanks for pointing out that we can just let express do it!